### PR TITLE
[PF-2593] Patch jersey2 post and put

### PR DESCRIPTION
**Pull request by rogerwangcs**
This change adds the jersey2 java api client template overrides to add an empty body to post and put requests to avoid null request body. Based off of https://github.com/DataBiosphere/terra-workspace-manager/pull/186.

See: https://broadworkbench.atlassian.net/browse/PF-2953

### CODE REVIEW

File: client/src/main/resources/swaggercodegen/libraries/jersey2/ApiClient.mustache [hunk 1]
```diff
@@ -0,0 +1,863 @@
...
+   * @param authNames The authentications to apply
+   * @param returnType The return type into which to deserialize the response
+   * @return The response body in type of string
+   * @throws ApiException API exception
+   */
+  public <T> T invokeAPI(String path, String method, List<Pair> queryParams, Object body, Map<String, String> headerParams, Map<String, Object> formParams, String accept, String contentType, String[] authNames, GenericType<T> returnType) throws ApiException {
+    updateParamsForAuth(authNames, queryParams, headerParams);
+
+    {{! Begin Terra change }}
+    if (body == null && ("POST".equals(method) || "PUT".equals(method))) {
```
Comment (rogerwangcs) [inline client/src/main/resources/swaggercodegen/libraries/jersey2/ApiClient.mustache:701]: The moustache templates are borrowed from WSM, this line is the only change from WSM.

File: client/src/main/resources/swaggercodegen/generatedAnnotation.mustache [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+  This file is not important to Terra's codegen customizations, but needs to be included in order
+  to get Swagger Codegen to successfully compile. If this file is not included, a Handlebars
+  exception is thrown ("The partial [path-to-generatedAnnotation.mustache] could not be found").
+
+  In theory, it should be possible to override individual Swagger template files without copying
+  all dependent templates locally (https://github.com/swagger-api/swagger-codegen/issues/10272),
+  but in practice it's necessary to include all files that the main template depends on.
+}}
+{{^hideGenerationTimestamp}}@javax.annotation.Generated(value = "{{generatorClass}}", date = "{{generatedDate}}"){{/hideGenerationTimestamp}}
```
Comment (dexamundsen) [inline client/src/main/resources/swaggercodegen/generatedAnnotation.mustache]: nit: new line

File: client/src/main/resources/swaggercodegen/generatedAnnotation.mustache [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{{!
+  Copied from the https://github.com/swagger-api/swagger-codegen-generators tagged release 1.0.24:
```
Comment (dexamundsen) [inline client/src/main/resources/swaggercodegen/generatedAnnotation.mustache:2]: Any reason to use 1.0.24 vs the latest 1.0.42? apples to other files as well

Comment (rogerwangcs) [reply client/src/main/resources/swaggercodegen/generatedAnnotation.mustache:2]: The swagger code gen dependency version is the same (1 patch behind) as WSM. Didn't want to risk having any side effects from bumping both.

**Review by dexamundsen — APPROVED** [review]

**Review by ddietterich — APPROVED** [review]
Looks OK to me. Does it work?

**Comment by rogerwangcs** [comment]
> Looks OK to me. Does it work?

I published the client to my local mavel repo, and sourced that from the my local terra cli. The empty body endpoints call without issues with the updated swagger client.

Here is the generated client code where the override is added:
Screenshot 2023-08-18 at 4 29 47 PM

**Pull request merged**